src/components/Speciality/Events.js

```diff
@@ -1,56 +1,71 @@
+import { format, isAfter, endOfYesterday } from 'date-fns'
+import { Padding } from 'styled-components-spacing'
+
+import specialityEventIcon from './assets/events-icon.svg'
 import { Row, Col, Grid } from '../grid'
 import { SectionTitle, Subtitle, BodyPrimary } from '../Typography'
 import ExternalAnchor from '../Common/ExternalAnchor'
-import { Padding } from 'styled-components-spacing'
-import { format } from 'date-fns'
+import Hr from '../Common/Hr'
 
-const EventBorder = styled(Col)`
-  border: 1px solid rgba(51, 51, 51, 0.15);
-`
+const noEventsMessage = title =>
+  `It looks like there currently aren’t any upcoming ${title} events. You can always check back again later or get in touch if you are interested in potentially hosting one.`
 
-const EventSection = ({ events, title, eventIcon }) => {
-  const futureEvents = (events || []).filter(
-    ({ startTime }) => new Date(startTime) > new Date()
+const isSpecialityEvent = (eventTitle, title) => {
+  const formattedTitle = title
+    .toLowerCase()
+    .replace(/(\.|js)/gi, '')
+    .trim()
+
+  return eventTitle.toLowerCase().includes(formattedTitle)
+}
+
+const getSpecialityEvents = (events = [], title) =>
+  events.filter(
+    ({ eventTitle, startTime }) =>
+      isSpecialityEvent(eventTitle, title) &&
+      isAfter(startTime, endOfYesterday())
   )
 
-  return futureEvents.length > 0 ? (
+const EventSection = ({ events, title, eventIcon }) => {
+  const specialityEvents = getSpecialityEvents(events, title)
+
+  return (
     <Grid>
       <Padding top={6} bottom={6}>
         <Row>
           <Col width={[1, 1, 1, 1, 6 / 12]}>
-            <SectionTitle>{`Upcoming ${title} events`}</SectionTitle>
+            <div>
+              <Padding bottom={1}>
+                <img src={specialityEventIcon} alt="events icon" />
+              </Padding>
+              <SectionTitle>{`Upcoming ${title} events`}</SectionTitle>
+            </div>
           </Col>
           <Col width={[1, 1, 1, 1, 6 / 12]}>
-            {futureEvents.map(event => (
-              <EventBorder key={`${event.id}`}>
-                <Padding top={2} bottom={2}>
-                  <Row>
-                    <Col>
-                      <img
-                        src={`https://${eventIcon.file.url}`}
-                        alt={eventIcon.title}
-                      />
-                    </Col>
-                    <Col>
-                      <Subtitle noPadding>
-                        <ExternalAnchor href={event.linkToEvent}>
-                          {event.eventTitle}
-                        </ExternalAnchor>
-                      </Subtitle>
-                      <BodyPrimary noPadding>
-                        {format(new Date(event.date), 'MMMM DD[,] dddd')}
-                      </BodyPrimary>
-                    </Col>
-                  </Row>
-                </Padding>
-              </EventBorder>
-            ))}
+            {specialityEvents.length ? (
```

Comment:
`{specialityEvents && specialityEvents.length && ....` specialityEvents could be null without `length` property

Reply:
In theory it shouldn't as I added a fall back so it equals to `[]`. I have a few updates to push so the code is more robust.